src/lib.rs

```diff
@@ -102,20 +103,21 @@ impl Cloudinary {
     /// let cloudinary = Cloudinary::new("api_key".to_string(), "cloud_name".to_string(), "api_secret".to_string() );
     /// let result = cloudinary.upload_image(Source::Path("./image.jpg".into()), &options);
     /// ```
-    pub async fn upload_image(
-        &self,
-        src: Source,
-        options: &UploadOptions<'_>,
-    ) -> Result<UploadResult> {
+
+    pub async fn upload(&self, src: Source, options: &UploadOptions<'_>) -> Result<UploadResult> {
         let client = Client::new();
         let file = match src {
             Source::Path(path) => prepare_file(&path).await?,
             Source::Url(url) => Part::text(url.as_str().to_string()),
         };
         let multipart = self.build_form_data(options).part("file", file);
         let url = format!(
-            "https://api.cloudinary.com/v1_1/{}/image/upload",
-            self.cloud_name
+            "https://api.cloudinary.com/v1_1/{}/{}/upload",
+            self.cloud_name,
+            options
+                .get_resource_type()
```

Comment:
I don't think upload options can be 100% between images, videos, and sound. For example, "faces" are relevant for images only.

Reply:
From my understanding of [the docs](https://cloudinary.com/documentation/upload_images#optional_upload_parameters), the upload parameters are somewhat general. Perhaps cloudinary's api will accept the valid options based on the resource_type or the  specific endpoint called, I'm nor really sure